The task.request is still sometimes "empty"

Despite 3f7b833f3, I am still getting this "sometimes":

```
billiard.einfo.ExceptionWithTraceback:
"""
Traceback (most recent call last):
  File "/main/srhaque/bootstrap/source/paiyroll/tasks.py", line 109, in viewflow_dequeue
    return CeleryResultCodec.encode_success(logger, celery_task, body(call_, viewflow_process, viewflow_task, celery_task))
  File "/main/srhaque/bootstrap/source/paiyroll/tasks.py", line 96, in body
    assert cl_task.request.id == vf_task.external_task_id, \
AssertionError: None != external task 09db61b7-f45b-4f1d-8752-6acadad2076e

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/celery_aio_pool/tracer.py", line 283, in trace_task
    R = retval = AsyncIOPool.run_in_pool(fun, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/celery_aio_pool/pool.py", line 222, in run_in_pool
    return worker_pool.run(
  File "/usr/local/lib/python3.10/dist-packages/celery_aio_pool/pool.py", line 203, in run
    raise error
  File "/usr/lib/python3.10/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/dist-packages/celery/app/trace.py", line 732, in __protected_call__
    return orig(self, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/celery/app/task.py", line 405, in __call__
    return self.run(*args, **kwargs)
  File "/main/srhaque/bootstrap/source/paiyroll/tasks.py", line 112, in viewflow_dequeue
    return CeleryResultCodec.encode_exception(logger, celery_task, e)
  File "/main/srhaque/bootstrap/source/paiyroll/xprocess.py", line 336, in encode_exception
    return cls._encode(logger, celery_states.FAILURE, celery_task, exception)
  File "/main/srhaque/bootstrap/source/paiyroll/xprocess.py", line 345, in _encode
    hostname = request.hostname.split('@', 1)[-1]
AttributeError: 'NoneType' object has no attribute 'split'
"""
```

As we discussed at one point, it seems there are multiple paths into the user's logic and at least one of these paths is clearing or recreating the `task.request`.

--- a/celery_aio_pool/__init__.py
+++ b/celery_aio_pool/__init__.py
@@ -34,6 +34,13 @@ def patch_celery_tracer() -> bool:
 
         celery.app.trace.build_tracer = build_async_tracer
 
+    #
+    # The "stack protections" installed by :func:`celery.app.trace.setup_worker_optimizations`
+    # cause the :class:`celery.Task` provided to bound task functions (i.e.
+    # decorated with "@celery_app.task(bind=True)" to be malformed.
+    # TODO: find out if this is the correct fix for that problem.
+    #
+    celery.app.trace.reset_worker_optimizations()
     return celery.app.trace.build_tracer is build_async_tracer
 
 

